src/ApiIncrementalChecker.ts

```diff
@@ -73,6 +77,32 @@ export class ApiIncrementalChecker implements IncrementalCheckerInterface {
     }
   }
 
+  private getLinterConfig(file: string): ConfigurationFile {
+    const dirname = path.dirname(file);
+    if (dirname in this.linterConfigs) {
+      return this.linterConfigs[dirname];
+    }
+    if (FsHelper.existsSync(path.join(dirname, 'tslint.json'))) {
+      const config = ApiIncrementalChecker.loadLinterConfig(
+        path.join(dirname, 'tslint.json')
+      );
+      if (config.linterOptions && config.linterOptions.exclude) {
+        this.linterExclusions.concat(
+          config.linterOptions.exclude.map(
+            pattern => new minimatch.Minimatch(path.resolve(pattern))
+          )
+        );
+      }
+      this.linterConfigs[dirname] = config;
+    } else {
+      if (dirname === '.') {
```

Comment:
can you talk me through what happens in this branch please?  I've read it a bunch of times but I'm not so clear how this copes with:

1. if there's only a `tslint.json` file in a particular directory of the application but *not* in the root
2. if this genuinely manages to walk up the directory tree.  When `this.linterConfigs[dirname] = this.getLinterConfig(dirname);` is invoked doesn't it stick with the current directory rather than walk up to the parent?  Maybe I'm misunderstanding this...

Reply:
1. Do you mean "What if I want to subject to tslint some particular parts of my project by providing `tslint.config` inside, for example, `src/components` folder, but not in root folder"? I assumed that it is not a legitimate case, because of the chains of imports inside `src/components` that could lead to files from the root of project, that is why I throw an error, correct me if I'm wrong
2. Follow these steps:
  - first file to be processed is `src/components/SomeComponent.tsx`. On the first run of `getLinterConfig` we get `const dirname = 'src/components'`
  - check if `src/components/tslint.json` exists (suppose it does not)
  - invoke second run of `getLinterConfig('src/components')`
  - `const dirname = path.dirname('src/components') = 'src'`
  - check `src/tslint.json`, if it does, then OK, we've got our config, if not - then invoke third run of `getLinterConfig('src')`
  - check `./tslint.json`, if it does not exist - then throw an error

Reply:
2. Cool - that's clear thanks

> I assumed that it is not a legitimate case, because of the chains of imports inside src/components that could lead to files from the root of project, that is why I throw an error, correct me if I'm wrong

I'm not sure it is an error...  How does tslint itself behave in those circumstances? My assumption is that in this case we should just not lint...

Reply:
yeah, you're right, TSLint CLI warns `No valid rules have been specified for TypeScript files` for root file, but still check inner files. Fixin' it